packages/docusaurus-theme-classic/src/theme/Toggle/index.tsx

```diff
@@ -13,17 +13,43 @@ import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
 import clsx from 'clsx';
 import styles from './styles.module.css';
 
-const Moon = () => <span className={clsx(styles.toggle, styles.moon)} />;
-const Sun = () => <span className={clsx(styles.toggle, styles.sun)} />;
+import {
+  DEFAULT_COLOR_MODE_CONFIG,
+  mergeDefault
+} from '../../validateThemeConfig';
+
+const Dark = ({icon, style}) => (
+  <span className={clsx(styles.toggle, styles.dark)} style={style}>
+    {icon}
+  </span>
+);
+const Light = ({icon, style}) => (
+  <span className={clsx(styles.toggle, styles.light)} style={style}>
+    {icon}
+  </span>
+);
 
 export default function (props: ComponentProps<typeof Toggle>): JSX.Element {
-  const {isClient} = useDocusaurusContext();
+  const {isClient, siteConfig = {}} = useDocusaurusContext();
+  const {
+    themeConfig: {
+      colorMode
+    },
+  } = siteConfig;
+  
+  const { switchConfig: {
+    darkIcon,
+    darkIconStyle,
+    lightIcon,
+    lightIconStyle
+  } } = mergeDefault(colorMode, DEFAULT_COLOR_MODE_CONFIG)
```

Comment:
the themeConfig should be provided to the components with default values already applied. We should not apply defaults in components